src/components/popups/popup_connect.gd

```diff
@@ -22,7 +26,7 @@ func _on_ButtonConnect_pressed() -> void:
 
 func _on_ConnectPopup_about_to_show() -> void:
 	psk_line_edit.text = API.psk
-	connect_to_core()
+	$ConnectDelay.start()
```

Comment:
should be declared on top as  onready like `status` and `psk_line_edit` for consistency? (same other lines below)

Reply:
There are other places in the script that use the `$` annotation in the script if they aren't used very often - I think it's fine for now 👍